spec/controllers/api/v1/media_controller_spec.rb

```diff
@@ -4,48 +4,148 @@
   let(:authorized_user) { create(:user) }
   let(:api_resource_name){ 'media' }
   let(:resource_class) { Medium }
-  let(:api_resource_attributes) { %w(id src media_type) }
+  let(:api_resource_attributes) { %w(id src media_type content_type created_at) }
   let(:api_resource_links) { [] }
   let(:scopes) { %w(public medium) }
 
-  RSpec.shared_examples "has_many media" do |parent_name, media_type|
-    describe "#index" do
-      let(:private_resource) { create(:project, private: true).avatar }
-      let(:n_visisble) { 2 }
+  RSpec.shared_examples "has_many media" do |parent_name, media_type, actions|
+    let!(:resources) do
+      create_list :medium, 2, linked: parent, content_type: "text/csv",
+        type: "#{parent_name}_#{media_type.to_s.singularize}"
+    end
+
+    if actions.include? :index
+      describe "#index" do
+        context "when media exists" do
+          before(:each) do
+            default_request user_id: authorized_user.id, scopes: scopes
+            get :index, :"#{parent_name}_id" => parent.id, :media_name => media_type
+          end
 
-      it_behaves_like 'is indexable'
+          it 'should return ok' do
+            expect(response).to have_http_status(:ok)
+          end
+
+          it 'should include 2 items' do
+            expect(json_response["media"].length).to eq(2)
+          end
+
+          it_behaves_like "an api response"
+        end
+
+        context "when media does not exist" do
+          let!(:resources) { [] }
+
+          before(:each) do
+            parent.send(media_type).destroy
```

Comment:
Yeah I think it would involve making a whole chain of stubs though, since some of of the instances of that destroy command are also doing an .exists? activerecord test.